src/Scene/LayerUpdateState.js

```diff
@@ -0,0 +1,57 @@
+const UPDATE_IDLE = 0;
+const UPDATE_PENDING = 1;
+const UPDATE_ERROR = 2;
+
+const PAUSE_BETWEEN_ERRORS = [ 1000.0, 3000.0, 7000.0, 60000.0 ];
+
+function LayerUpdateState() {
+    this.state = UPDATE_IDLE;
+    this.lastErrorTimestamp = 0;
+    this.errorCount = 0;
+
+}
+
+LayerUpdateState.prototype.constructor = LayerUpdateState;
+
+LayerUpdateState.prototype.canTryUpdate = function(timestamp) {
+    switch (this.state) {
+        case UPDATE_IDLE: {
+            return true;
+        }
+        case UPDATE_PENDING: {
+            return false;
+        }
+        case UPDATE_ERROR:
+        default: {
+            let errorDuration = this.waitDurationUntilNextTry();
+            return errorDuration <= (timestamp - this.lastErrorTimestamp);
+        }
+    }
+}
+
+LayerUpdateState.prototype.waitDurationUntilNextTry = function() {
```

Comment:
I don't use `get`  for method that aren't accessors.
But I agree with your comment so I changed the name to: `secondsUntilNextTry()`